packages/create-cloudflare/src/templateMap.ts

```diff
@@ -1,53 +1,368 @@
+ * function adapts a remote template to work with c3 by inferring a simple config from
+ * its file structure.
+ */
+const inferTemplateConfig = (path: string): TemplateConfig => {
+	return {
+		configVersion: 1,
+		id: "remote-template",
+		displayName: "A remote C3 template",
+		platform: "workers",
+		copyFiles: inferCopyFilesDefinition(path),
+	};
+};
+
+const inferCopyFilesDefinition = (path: string) => {
+	const copyFiles: StaticFileMap | VariantInfo = {};
+	if (existsSync(join(path, "js"))) {
+		copyFiles["js"] = { path: "./js" };
+	}
+	if (existsSync(join(path, "ts"))) {
+		copyFiles["ts"] = { path: "./ts" };
+	}
+	if (Object.keys(copyFiles).length !== 0) {
+		return copyFiles;
+	}
+
+	return {
+		path: ".",
+	};
+};
+
+/**
+ * Downloads an external template from a git repo using `degit`.
+ *
+ * @param src The url of the git repository to download the template from.
+ *            For convenience, `owner/repo` is also accepted.
+ * @returns A path to a temporary directory containing the downloaded template
+ */
+const downloadRemoteTemplate = async (src: string) => {
+	try {
+		const s = spinner();
+		s.start(`Cloning template from: ${blue(src)}`);
+		const emitter = degit(src, {
+			cache: false,
+			verbose: false,
+			force: true,
+		});
+
+		const tmpDir = await mkdtemp(join(tmpdir(), "c3-template"));
+		await emitter.clone(tmpDir);
+		s.stop(`${brandColor("template")} ${dim("cloned and validated")}`);
+
+		return tmpDir;
+	} catch (error) {
+		return crash(`Failed to clone remote template: ${src}`);
+	}
+};
+
+export const updatePackageJson = async (ctx: C3Context) => {
+	const s = spinner();
+	s.start("Updating `package.json`");
+
+	// Update package.json with project name
+	const placeholderNames = ["<TBD>", "TBD", ""];
+	const pkgJsonPath = resolve(ctx.project.path, "package.json");
+	let pkgJson = readJSON(pkgJsonPath);
+
+	if (placeholderNames.includes(pkgJson.name)) {
+		pkgJson.name = ctx.project.name;
+	}
+
+	// Run any transformers defined by the template
+	if (ctx.template.transformPackageJson) {
+		const transformed = await ctx.template.transformPackageJson(pkgJson);
+		pkgJson = deepmerge(pkgJson, transformed);
+	}
+
+	// Write the finalized package.json to disk
+	writeJSON(pkgJsonPath, pkgJson);
+	s.stop(`${brandColor("updated")} ${dim("`package.json`")}`);
+};
+
+export const getTemplatePath = (ctx: C3Context) => {
+	if (ctx.template.path) {
+		return ctx.template.path;
+	}
+
+	return resolve(__dirname, "..", "templates", ctx.template.id);
+};
+
+// const readTemplateConfig = async (templatePath: string) => {
+// 	const resolvedTemplatePath = resolve(__dirname, templatePath);
+// 	const importPath = probePaths(
+// 		[
+// 			join(resolvedTemplatePath, "c3.json"),
+// 			join(resolvedTemplatePath, "c3.ts"),
+// 			join(resolvedTemplatePath, "c3.js"),
+// 		],
+// 		`Failed to find a c3 template config file at: ${templatePath}.`
+// 	);
+
+// 	return await require(importPath);
+// };
```

Comment:
I'm not the biggest fan of keeping unused commented-out code in the codebase 😕

```suggestion
```

If you really really really want to keep this in I think there should at least be a comment explaining this
(although again, my preference would be to get rid of this entirely, re-implementing it if/when needed shouldn't be too hard)

Reply:
Nope, I can remove